File: internal/infra/aws/dynamodb_client.go
package infra_aws

import (
	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
)

type DynamoDBClient struct {
	*dynamodb.Client
}

func NewDynamoDBClient(region AwsRegion, stage AwsStage) *DynamoDBClient {
	if stage != StageLocal {
		return &DynamoDBClient{
			Client: dynamodb.NewFromConfig(NewConfig(region)),
		}
	}

	return &DynamoDBClient{
		Client: dynamodb.NewFromConfig(NewLocalConfig(region)),
	}
}

File: src/infra/aws/dynamodb_client/dynamodb_client.ts
import { DynamoDBClient } from "@aws-sdk/client-dynamodb"
import { AwsRegion, AwsStage, newAwsConfig } from "../utils"

export class DynamoDBClientWrapper extends DynamoDBClient {
	constructor(region: AwsRegion, stage: AwsStage) {
		const config = newAwsConfig(region, stage)
		super(config)
	}
}